### PR TITLE
cfonts 1.0.2 (new formula)

### PR DESCRIPTION
**Pull request by bladey**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/cfonts.rb [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+      system "make"
+      system "cargo", "install", *std_cargo_args
+      bin.install "target/release/cfonts"
+    end
+  end
+
+  test do
+    system bin/"cfonts", "--version"
+    assert_match "\n\n ████████╗\n ╚══██╔══╝\n    ██║   \n    ██║   \n    ██║   \n    ╚═╝   \n\n\n",
+      shell_output("#{bin}/cfonts t")
```
Comment (carlocab) [inline Formula/cfonts.rb:21]:
```suggestion
    assert_match <<~EOS, shell_output("#{bin}/cfonts t")


    \ ████████╗
    \ ╚══██╔══╝
    \    ██║  \s
    \    ██║  \s
    \    ██║  \s
    \    ╚═╝  \s


    EOS
```
I may have gotten the whitespace wrong somewhere, but you get the idea.

File: Formula/cfonts.rb [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    assert_match "\n\ntest\n\n\n",
+      shell_output("#{bin}/cfonts test -f console")
```
Comment (carlocab) [inline Formula/cfonts.rb:23]:
```suggestion
    assert_match "\n\ntest\n\n\n", shell_output("#{bin}/cfonts test -f console")
```

File: Formula/cfonts.rb [hunk 3]
```diff
@@ -0,0 +1,25 @@
+class Cfonts < Formula
```
Comment (carlocab) [inline Formula/cfonts.rb:1]: Shouldn't we indicate that this is the Rust implementation of `cfonts`?

**Comment by carlocab** [comment]
By the way, for future reference, we put the name of the formula first in our commit messages, like so:
```
cfonts 1.0.2 (new formula)
```
Please see the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit) referenced in the PR template for details.

File: Formula/cfonts.rb (hunk 3 above)
Comment (dominikwilkowski) [reply Formula/cfonts.rb:1]: Hey, I'm the author and I feel like node isn't a great way to distribute cli tools which is why I wrote the rust port. So there will never be a nodejs implementation distributes anywhere other than yarn and npm. Does that help?

File: Formula/cfonts.rb (hunk 1 above)
Comment (dominikwilkowski) [reply Formula/cfonts.rb]: Looks like this fails the with linter error: https://github.com/Homebrew/homebrew-core/runs/6652520254?check_suite_focus=true#step:4:24

Happy to revert or is there another nicer way @carlocab?

**Comment by dominikwilkowski** [comment]
Thank you @carlocab for the comments and help. It should all be aligned now. Let me know if I can help further.

File: Formula/cfonts.rb (hunk 3 above)
Comment (carlocab) [reply Formula/cfonts.rb:1]: We distribute just over a hundred CLI tools that are installed via `npm`, so I'm not sure we can guarantee that there will never be a distribution of the nodejs implementation other than through `yarn` or `npm`.

Comment (dominikwilkowski) [reply Formula/cfonts.rb:1]: If someone was to distribute my nodejs library through brew against my consent than at least it should be called `cfonts-nodejs` or something no?

Comment (dominikwilkowski) [reply Formula/cfonts.rb:1]: The way I phrased my first comment was a bit confusing I admit. I am not saying brew is a bad way to distribute node scripts. I love brew and use it daily. What I meant was that the nodejs implementation of `cfonts` is not as good as the rust one and I would rather publish rust on here than the nodejs one. Hope that's more clear

**Review by SMillerDev — APPROVED** [review]

**Review by chenrui333 — APPROVED** [review]

File: Formula/cfonts.rb (hunk 3 above)
Comment (dominikwilkowski) [reply Formula/cfonts.rb:1]: @carlocab Is there anything I can do to unblock this PR?

**Pull request closed without merging**